source/common/common/logger.h

```diff
@@ -123,48 +124,51 @@ template <Id id> class Loggable {
 
 } // Logger
 
+/**
+ * Base logging macros.
+ */
 #ifdef NDEBUG
-#define log_trace(...)
-#define log_debug(...)
+#define log_trace(LOGGER, ...)
+#define log_debug(LOGGER, ...)
 #else
-#define log_trace(...) log().trace(__VA_ARGS__)
-#define log_debug(...) log().debug(__VA_ARGS__)
+#define log_trace(LOGGER, ...) LOGGER.trace(__VA_ARGS__)
+#define log_debug(LOGGER, ...) LOGGER.debug(__VA_ARGS__)
 #endif
 
+#define log_info(LOGGER, ...) LOGGER.info(__VA_ARGS__)
+#define log_warn(LOGGER, ...) LOGGER.warn(__VA_ARGS__)
+#define log_err(LOGGER, ...) LOGGER.err(__VA_ARGS__)
+#define log_critical(LOGGER, ...) LOGGER.critical(__VA_ARGS__)
+
 /**
- * Convenience macros for logging with connection ID.
+ * Convenience macros to call the above macros with or without a logger.
  */
-#define conn_log(LOG, LEVEL, FORMAT, CONNECTION, ...)                                              \
-  LOG.LEVEL("[C{}] " FORMAT, (CONNECTION).id(), ##__VA_ARGS__)
+#define log_facility(LEVEL, ...) log_##LEVEL(log(), ##__VA_ARGS__)
+#define log_to_logger(LOGGER, LEVEL, ...) log_##LEVEL(LOGGER, ##__VA_ARGS__)
 
-#ifdef NDEBUG
-#define conn_log_trace(...)
-#define conn_log_debug(...)
-#else
-#define conn_log_trace(FORMAT, CONNECTION, ...)                                                    \
-  conn_log(log(), trace, FORMAT, CONNECTION, ##__VA_ARGS__)
-#define conn_log_debug(FORMAT, CONNECTION, ...)                                                    \
-  conn_log(log(), debug, FORMAT, CONNECTION, ##__VA_ARGS__)
-#endif
+/**
+ * Convenience macros to log to the general logger.
+ */
+#define get_general_logger() Logger::Loggable<Logger::Id::general>::log()
+#define log_general(LEVEL, ...) log_to_logger(get_general_logger(), LEVEL, ##__VA_ARGS__)
```

Comment:
Should this be the `general` logger or `misc` logger. I think if it was called misc it would cause folks to think twice before using it when there is a reasonable way to put this in a "facility".